File: sdk/go/common/util/cmdutil/testdata/term_graceful.py
import signal
import sys
import time

def signal_handler(signal, frame):
    print("exiting cleanly")
    sys.exit(0)

signal.signal(signal.SIGINT, signal_handler)
if hasattr(signal, "SIGBREAK"):
    # SIGBREAK is only available on Windows
    signal.signal(signal.SIGBREAK, signal_handler)
print("ready", flush=True)

time.sleep(3)
print("error: signal not received", file=sys.stderr)
sys.exit(1)

File: sdk/go/common/util/cmdutil/testdata/term_graceful.go
//go:build ignore
// +build ignore

package main

import (
	"fmt"
	"log"
	"os"
	"os/signal"
	"time"
)

func main() {
	log.SetFlags(0)

	sigch := make(chan os.Signal, 1)
	signal.Notify(sigch, os.Interrupt)
	// os.Interrupt handles SIGINT and CTRL_BREAK_EVENT.

	fmt.Println("ready")
	select {
	case <-sigch:
		log.Println("exiting cleanly")
		os.Exit(0)

	case <-time.After(3 * time.Second):
		log.Fatal("error: did not receive signal")
	}
}
